### PR TITLE
re #373 & #374 make all plugin configs dynamic

### PR DESCRIPTION
**Pull request by zeroasterisk**
I did this mostly with a macro, and a bit of hand cleanup - but truthfully I kinda flew through it, and use little of it - please review

### CODE REVIEW

**Comment by zeroasterisk** [comment]
It looks like at least some of this isn't right yet....
ss

**Comment by wsdjeg** [comment]
@zeroasterisk when do you get this errors?

**Comment by zeroasterisk** [comment]
when typing --- I found a problem with some of my `get()` where the keys had `.` in the path, but my getter didn't... reverting those lines....

**Comment by zeroasterisk** [comment]
is this valid?
`let g:deoplete#ignore_sources.rust = get(g:, 'deoplete#ignore_sources.rust', ['omni'])`


if not, i can just revert if they have dot in the key path... if so, I can fix...

**Comment by wsdjeg** [comment]
I will test it now.

**Comment by wsdjeg** [comment]
no, it is not valid.
let g:deoplete#ignore_sources.rust = get(g:deoplete#ignore_sources, 'rust', ['omni'])

**Comment by wsdjeg** [comment]
`rust` is a key in `g:deoplete#ignore_sources`

**Comment by zeroasterisk** [comment]
so this should work:
`let g:deoplete#ignore_sources._ = get(g:, 'deoplete#ignore_sources', '_', ['around'])`

**Comment by wsdjeg** [comment]
no, get() func only support 3 argv
so it should be
`let g:deoplete#ignore_sources._ = get(g:deoplete#ignore_sources,  '_',  ['around'])`

**Comment by zeroasterisk** [comment]
^ that seems to have fixed it - from what I can see

File: config/plugins/denite.vim [hunk 1]
```diff
@@ -13,7 +14,7 @@ let s:denite_options = {'default' : {
       \ 'direction': 'rightbelow',
       \ 'statusline' : 'false',
       \ 'prompt' : '➭',
-      \ }}
+      \ }})
```
Comment (wsdjeg) [inline config/plugins/denite.vim]: no need change s:value, this will not be changed by users.

Comment (zeroasterisk) [reply config/plugins/denite.vim]: reverted

**Comment by wsdjeg** [comment]
LGTM, 👍

**Pull request merged**